### PR TITLE
Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by nickelc**
The reason to choose `setup-rust-action` over `action-rs/*` is that the action already comes with problem matchers for error, warning & cargo. This makes `actions-rs/cargo` obsolete and the workflow less verbose.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    name: Test
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        rust: [stable, beta, nightly]
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: hecrj/setup-rust-action@v1
```
Comment (seanmonstar) [inline .github/workflows/ci.yml:19]: Any reason not to use `actions-rs/..`?

Comment (nickelc) [reply .github/workflows/ci.yml:19]: Sorry, for being :sloth: . I've updated the PR comment.

**Comment by seanmonstar** [comment]
Thanks for explaining some. The reason I ask is that unless there's a significant reason otherwise, it makes sense to me to keep things similar across hyper, h2, http, etc. Does that seem reasonable? Is there places in the Rust ecosystem that are shifting from `actions-rs` to this one?

**Comment by nickelc** [comment]
I think the `setup-rust-action` action came first but here we go now with `actions-rs/*`.

**Pull request merged**